src/shims/windows/thread.rs

```diff
@@ -66,7 +66,12 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         let timeout = this.read_scalar(timeout_op)?.to_u32()?;
 
         let thread = match Handle::from_scalar(handle, this)? {
-            Some(Handle::Thread(thread)) => thread,
+            Some(Handle::Thread(thread)) => {
+                match this.thread_id_try_from(thread) {
+                    Ok(thread) => thread,
+                    Err(_) => return interp_ok(0xFFFFFFFF), // WAIT_FAILED
```

Comment:
The docs seem to say that this also sets the last-error variable.

Reply:
That's not documented too, so I will make a test app tomorrow. Or maybe will find something in [the definitive list of the error codes](https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes).

Reply:
Maybe @ChrisDenton knows

Reply:
That would be `ERROR_INVALID_HANDLE`.